CPP_STYLE_GUIDE.md

```diff
@@ -1,5 +1,23 @@
 # C++ Style Guide
 
+## Table of Contents
+
+* [Left-leaning (C++ style) asterisks for pointer declarations](#left-leaning-c-style-asterisks-for-pointer-declarations)
```

Comment:
Not really need to be done in this PR, but we should probably put these rules into different categories so they would be easier to look up

Reply:
Oh, I think https://github.com/nodejs/node/pull/17052 is actually adding ToC there, maybe this PR should not include that change?

Reply:
That's kind of why I added a Table of Contents, so folks get an idea of all the rules even if they don't scroll all the way to the bottom. If anybody has a good idea how to order them, I'm all for it.

Reply:
Ideally, we have a linter that enforces all the rules 🤣

Reply:
@fhinkel Yes, but an easy-to-extend and easy-to-use C++ linter is not easy to come by as their JavaScript counterparts.. I've been looking into clang-tidy while working on https://github.com/nodejs/node/pull/16122, but that requires a compilation database, we do have ninja support that can emit those, still it probably raise the barrier of entry too high for new contributors..